include/kernel/thread.h

```diff
@@ -309,3 +285,29 @@ void awaken(void);
 void postpone(sched_context_t *sc);
 #endif
 
+/* Update the kernel's timestamp and store it in ksCurTime.
+ * The difference between the previous kernel timestamp and the one just read
+ * is added to ksConsumed.
+ *
+ * Should be called on every kernel entry where threads can be billed.
+ * Will be called without holding the kernel lock!
+ */
+static inline void updateTimestamp(void)
+{
+#ifdef CONFIG_KERNEL_MCS
+    time_t prev = NODE_STATE(ksCurTime);
+    NODE_STATE(ksCurTime) = getCurrentTime();
+    assert(NODE_STATE(ksCurTime) < MAX_RELEASE_TIME);
+    time_t consumed = (NODE_STATE(ksCurTime) - prev);
+    NODE_STATE(ksConsumed) += consumed;
+    /* Not supported for SMP: */
```

Comment:
I wonder, now that we call `updateTimestamp()` on every kernel entry, would it be worth making the `numDomain` check compiled in conditionally at non-SMP only? Might save a few instructions everytime.